partiql-ast/src/ast.rs

```diff
@@ -965,6 +971,14 @@ pub enum Type {
     CustomType(CustomType),
 }
 
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct TimeType {
+    pub precision: Option<u32>,
+    pub tz: bool,
```

Comment:
nit: I think `tz` may not be descriptive enough to represent if a `TIME` has or doesn't have a timezone. Perhaps `withTz` is more descriptive?